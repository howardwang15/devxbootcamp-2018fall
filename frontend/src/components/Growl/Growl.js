import React from 'react';
import dateFormat from 'dateformat';

import './Growl.scss';

const Growl = ({id, userId, userName, text, createdAt}) => (
  <div className="growl-card">
    <div className="growl-meta">
      <p className="growl-user-name">{userName}</p>
      {createdAt && (
        <p className="growl-created-at">
          {dateFormat(createdAt, 'mmm d, yyyy h:mm tt')}
        </p>
      )}
    </div>
    <p className="growl-text">{text}</p>
  </div>
);

export default Growl;
