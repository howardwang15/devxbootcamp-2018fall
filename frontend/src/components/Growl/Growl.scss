@import 'style/color';

.growl-card {
  padding: 10px;

  .growl-text {
    font-size: 20px;
    color: $color-text-dark-1;
    line-height: 27.4px;
  }

  p {
    font-size: 16px;
    color: $color-text-dark-3;

    &.growl-user-name {
      font-weight: bold;
    }

    &.growl-text {
      color: $color-text-dark-1;
    }
  }

  .growl-meta {
    display: flex;
    & > * {
      margin-right: 4px;
    }
  }
}
