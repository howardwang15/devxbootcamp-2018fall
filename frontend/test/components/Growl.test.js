import React from 'react';
import renderer from 'react-test-renderer';

import Growl from '../../src/components/Growl';

describe('Growl component', () => {
  it('renders as expected', () => {
    const props = {
      id: 1,
      userId: 2,
      text: 'Test text',
      createdAt: new Date(
        'Sat Oct 06 2018 13:09:47 GMT-0700 (Pacific Daylight Time)',
      ),
    };
    const tree = renderer.create(<Growl {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
