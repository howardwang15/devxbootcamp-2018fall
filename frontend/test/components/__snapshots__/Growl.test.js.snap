// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Growl component renders as expected 1`] = `
<div
  className="growl-card"
>
  <div
    className="growl-meta"
  >
    <p
      className="growl-user-name"
    />
    <p
      className="growl-created-at"
    >
      Oct 6, 2018 1:10 pm
    </p>
  </div>
  <p
    className="growl-text"
  >
    Test text
  </p>
</div>
`;
